### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by vladimirmomiroski**
PR ready for review

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+.idea
+.vscode
+
 # See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
 
 # dependencies
```
Comment (AlexMelov) [inline .gitignore:6]: Remove comments

File: .gitignore [hunk 2]
```diff
@@ -13,6 +16,7 @@
 
 # misc
```
Comment (AlexMelov) [inline .gitignore:17]: Remove comments

File: README.md [hunk 3]
```diff
@@ -1 +1,59 @@
...
+```
+npm install --prefix ./backend
+```
+```
+npm install --prefix ./frontend
+```
+
+## Setup
+
+Create a .env.local file inside frontend directory:
```
Comment (AlexMelov) [inline README.md:24]: Provide example of database link without credentials if you are using mongo example of the link 
DB_URL = mongo<user><password>

File: backend/.eslintrc [hunk 4]
```diff
@@ -0,0 +1,19 @@
+{
+  "env": {
```
Comment (AlexMelov) [inline backend/.eslintrc:2]: put bracket { on new line it should look like 
``` 
"env" :
{
    "es6": true,
    "node": true
},

File: backend/.eslintrc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    "es6": true,
+    "node": true
+  },
+  "extends": ["@isnotdefined/eslint-config"],
+  "parserOptions": {
```
Comment (AlexMelov) [inline backend/.eslintrc:7]: brackets { on new line

File: backend/.eslintrc [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "sourceType": "module",
+    "ecmaVersion": 2018
+  },
+
+  "rules": {
```
Comment (AlexMelov) [inline backend/.eslintrc:12]: { on new line

File: backend/package.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  "main": "App.js",
+  "dependencies": {
+    "config": "3.3.1",
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.0",
+    "express": "^4.17.3",
+    "mongoose": "^6.3.0",
+    "nodemon": "^2.0.15"
+  },
+  "devDependencies": {
```
Comment (AlexMelov) [inline backend/package.json:13]: { on new line

File: backend/package.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    "@isnotdefined/eslint-config": "^4.8.0",
+    "eslint": "^8.14.0",
+    "prettier": "^2.6.2"
+  },
+  "scripts": {
```
Comment (AlexMelov) [inline backend/package.json:18]: { on new line

File: backend/src/Controllers/deleteTodoHandler.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+	{
+		const removedTodo = await Todo.deleteOne({ _id: id });
+
+    response.json(removedTodo);
+	}
+	catch (error) 
+	{
+    response.json({ message: error });
+	}
+};
```
Comment (AlexMelov) [inline backend/src/Controllers/deleteTodoHandler.js:17]: don't use try catch, try to chain them

File: backend/src/Routes/routes.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+const express = require('express');
```
Comment (AlexMelov) [inline backend/src/Routes/routes.js:1]: add space between variables and functions to be cleaner

File: backend/src/index.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
+const { mongoose } = require('mongoose');
+const express = require('express');
+const cors = require('cors');
+
+const app = express();
+
+app.use(express.json());
+
+// Middlewares
```
Comment (AlexMelov) [inline backend/src/index.js:9]: remove comments

File: backend/src/index.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+
+// routes
+
+app.use('/', require('./Routes/routes'));
+app.use('/todos', require('./Routes/routes'));
+app.use('/todos/:id', require('./Routes/routes'));
+
+// db connection
+mongoose
+  .connect('mongodb+srv://vladimir:R8QqnGKLQrEW1F8v@cluster0.8wijq.mongodb.net/todo_app?retryWrites=true&w=majority', {
```
Comment (AlexMelov) [inline backend/src/index.js:20]: remove credential from the file,  credentials should be saved in .env file which will not be provided on github. This way everybody can have a access to your database.

File: backend/src/index.js [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+  })
+  .catch(error => 
+  {
+  	throw new Error(error);
+  });
+
+// listening to the server
```
Comment (AlexMelov) [inline backend/src/index.js:29]: remove comment

File: backend/src/models/Todos.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+
+const TodoSchema = mongoose.Schema({
```
Comment (AlexMelov) [inline backend/src/models/Todos.js:3]: bracket { in new line

File: backend/src/models/Todos.js [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+  name: {
```
Comment (AlexMelov) [inline backend/src/models/Todos.js:4]: bracket { in new line

File: backend/src/models/Todos.js [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+    type: String,
+    required: true
+  },
+  isCompleted: {
```
Comment (AlexMelov) [inline backend/src/models/Todos.js:8]: { in new line

File: frontend/.eslintrc [hunk 17]
```diff
@@ -0,0 +1,24 @@
+{
+	"env": {
```
Comment (AlexMelov) [inline frontend/.eslintrc:2]: { on new line

File: frontend/.eslintrc [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+		"es6": true,
+		"node": true
+	},
+	"extends": ["@isnotdefined/eslint-config", "react-app", "react-app/jest"],
+	"parserOptions": {
```
Comment (AlexMelov) [inline frontend/.eslintrc:7]: { on new line

File: frontend/.eslintrc [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+		"sourceType": "module",
+		"ecmaFeatures": {
```
Comment (AlexMelov) [inline frontend/.eslintrc:9]: { on new line

File: frontend/.eslintrc [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+			"jsx": true
+		},
+		"ecmaVersion": 2018
+	},
+
+	"rules": {
```
Comment (AlexMelov) [inline frontend/.eslintrc:15]: { on new line

File: frontend/package.json [hunk 21]
```diff
@@ -1,38 +1,33 @@
 {
-  "name": "frontend",
-  "version": "0.1.0",
-  "private": true,
-  "dependencies": {
```
Comment (AlexMelov) [inline frontend/package.json:5]: { on new line

File: frontend/package.json [hunk 22]
```diff
@@ -1,38 +1,33 @@
...
-    "@testing-library/jest-dom": "^5.16.4",
-    "@testing-library/react": "^13.1.1",
-    "@testing-library/user-event": "^13.5.0",
-    "react": "^18.0.0",
-    "react-dom": "^18.0.0",
-    "react-scripts": "5.0.1",
-    "web-vitals": "^2.1.4"
-  },
-  "scripts": {
```
Comment (AlexMelov) [inline frontend/package.json:14]: { on new line

File: frontend/package.json [hunk 23]
```diff
@@ -1,38 +1,33 @@
...
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test",
-    "eject": "react-scripts eject"
-  },
-  "eslintConfig": {
-    "extends": [
```
Comment (AlexMelov) [inline frontend/package.json:21]: { & [ on new line

File: frontend/package.json [hunk 24]
```diff
@@ -1,38 +1,33 @@
...
-      "react-app",
-      "react-app/jest"
-    ]
-  },
-  "browserslist": {
-    "production": [
```
Comment (AlexMelov) [inline frontend/package.json:27]: { on new line

File: frontend/package.json [hunk 25]
```diff
@@ -1,38 +1,33 @@
...
-      ">0.2%",
-      "not dead",
-      "not op_mini all"
-    ],
-    "development": [
```
Comment (AlexMelov) [inline frontend/package.json:32]: [ on new line

File: frontend/package.json [hunk 26]
```diff
@@ -1,38 +1,33 @@
...
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
-  }
+	"name": "frontend",
+	"version": "0.1.0",
+	"private": true,
+	"proxy": "http://localhost:5000",
+	"dependencies": {
```
Comment (AlexMelov) [inline frontend/package.json:6]: { on new line

File: frontend/package.json [hunk 27]
```diff
@@ -1,38 +1,33 @@
...
+		"@testing-library/jest-dom": "^5.16.4",
+		"@testing-library/react": "^13.1.1",
+		"@testing-library/user-event": "^13.5.0",
+		"harmonize.css": "^4.1.4",
+		"react": "^18.0.0",
+		"react-dom": "^18.0.0",
+		"react-scripts": "5.0.1",
+		"web-vitals": "^2.1.4"
+	},
+	"devDependencies": {
```
Comment (AlexMelov) [inline frontend/package.json:16]: { on new line

File: frontend/package.json [hunk 28]
```diff
@@ -1,38 +1,33 @@
...
+		"@isnotdefined/eslint-config": "^4.8.0",
+		"eslint": "^8.14.0",
+		"eslint-config-stylelint": "^15.0.0",
+		"eslint-plugin-react": "^7.29.4",
+		"prettier": "^2.6.2",
+		"stylelint": "^14.8.0",
+		"@isnotdefined/stylelint-config": "^3.2.0"
+	},
+	"scripts": {
```
Comment (AlexMelov) [inline frontend/package.json:25]: { on new line

File: frontend/src/App.js [hunk 29]
```diff
@@ -1,13 +1,12 @@
-import './App.css';
-import React from 'react'
-
-
-function App() {
```
Comment (AlexMelov) [inline frontend/src/App.js:5]: { on new line

File: frontend/src/Todo/Body/BodyItemActions.jsx [hunk 30]
```diff
@@ -0,0 +1,18 @@
...
+export default function TodoBodyItemActions({ id }) 
+{
+	const { deleteBtn } = classes;
+
+	const { deleteTodo } = useContext(Context);
+
+	return (
+		<div>
+			<button onClick={() => deleteTodo(id)} className={deleteBtn}>
+        Delete
```
Comment (AlexMelov) [inline frontend/src/Todo/Body/BodyItemActions.jsx:14]: this Delete either put him together with button tag like <button>Delete</button> or correct the indent

File: frontend/src/Todo/Body/BodyItemCheckbox.jsx [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+import { Context } from '../Context';
+
+export default function TodoBodyItemCheckbox({ item }) 
+{
+	const { container, nameText, checkmark, checked } = classes;
+	const { _id, name, isCompleted } = item;
+
+	const { checkTodoAsCompleted } = useContext(Context);
+
+	const checkCompleted = isCompleted ? checked : '';
```
Comment (AlexMelov) [inline frontend/src/Todo/Body/BodyItemCheckbox.jsx:12]: don't use empty string, use null

File: frontend/src/Todo/Context.js [hunk 32]
```diff
@@ -0,0 +1,44 @@
...
+			sendMethodToServer('DELETE', id, setTodoData);
+		}
+	};
+
+	const checkTodoAsCompleted = id => 
+	{
+		sendMethodToServer('PATCH', id, setTodoData);
+	};
+
+	const contextObj = {
```
Comment (AlexMelov) [inline frontend/src/Todo/Context.js:36]: { on new line

File: frontend/src/Todo/Context.js [hunk 33]
```diff
@@ -0,0 +1,44 @@
...
+    todoData,
+    addTodo,
+    deleteTodo,
+    checkTodoAsCompleted
```
Comment (AlexMelov) [inline frontend/src/Todo/Context.js:40]: use tabs not spaces for indent

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 34]
```diff
@@ -0,0 +1,62 @@
...
+      })
+      .catch(error => 
+      {
+      	throw new Error(error);
+      });
+};
+
+export const fetchPost = (item, setTodoData) => 
+{
+	const options = {
```
Comment (AlexMelov) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:19]: { on new line

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 35]
```diff
@@ -0,0 +1,62 @@
...
+      method: 'POST',
+      headers: {
```
Comment (AlexMelov) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:21]: { on new line

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 36]
```diff
@@ -0,0 +1,62 @@
...
+    })
+    .catch(error => 
+    {
+    	throw new Error(error);
+    });
+};
+
+export const sendMethodToServer = (method, id, setTodoData) => 
+{
+	const options = {
```
Comment (AlexMelov) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:43]: { on new line

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 37]
```diff
@@ -0,0 +1,62 @@
...
+      method,
+      headers: {
```
Comment (AlexMelov) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:45]: { on new line

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 38]
```diff
@@ -0,0 +1,62 @@
...
+        'Content-Type': 'application/json'
+      }
+    };
+
+    fetch(`${urlToFetch}/${id}`, options)
+      .then(res => 
+      {
+      	if (res.status === 200)
```
Comment (AlexMelov) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:53]: use tabs

File: frontend/src/Todo/Header/Header.jsx [hunk 39]
```diff
@@ -0,0 +1,42 @@
...
+export default function TodoHeader() 
+{
+	const { container, input, btn } = classes;
+
+	const { addTodo } = useContext(Context);
+
+	const [ todoName, setTodoName ] = useState('');
+
+	const onSubmitHandler = e => 
+	{
```
Comment (AlexMelov) [inline frontend/src/Todo/Header/Header.jsx:14]: wrong indent

File: frontend/src/Todo/Header/Header.jsx [hunk 40]
```diff
@@ -0,0 +1,42 @@
...
+    	const id = new Date().valueOf();
+    	const todoItem = {
+        id,
+        isCompleted: false,
+        name: todoName
+      };
+
+    	addTodo(todoItem);
+    	setTodoName('');
+    }
```
Comment (AlexMelov) [inline frontend/src/Todo/Header/Header.jsx:27]: Wrong indent, you may be confused with assign id like this. If you are using mongoDb, mogno assign id itself on every todo.

File: frontend/src/Todo/Header/Header.jsx [hunk 41]
```diff
@@ -0,0 +1,42 @@
...
+import classes from './Header.module.css';
+import { Context } from '../Context';
+
+export default function TodoHeader() 
+{
+	const { container, input, btn } = classes;
+
+	const { addTodo } = useContext(Context);
+
+	const [ todoName, setTodoName ] = useState('');
```
Comment (AlexMelov) [inline frontend/src/Todo/Header/Header.jsx:11]: don't use empty strings

File: package.json [hunk 42]
```diff
@@ -0,0 +1,27 @@
+{
+	"name": "todo_app",
+	"author": "Vladimir Momiroski",
+	"description": "Todo Application",
+	"homepage": "https://github.com/vladimirmomiroski/todo_app#readme",
+	"version": "1.0.0",
+	"main": "index.js",
+	"license": "ISC",
+	"dependencies": {
```
Comment (AlexMelov) [inline package.json:9]: { on new line

File: package.json [hunk 43]
```diff
@@ -0,0 +1,27 @@
...
+		"husky": "^7.0.4"
+	},
+	"repository": {
```
Comment (AlexMelov) [inline package.json:12]: { on new line

File: package.json [hunk 44]
```diff
@@ -0,0 +1,27 @@
...
+		"type": "git",
+		"url": "git+https://github.com/vladimirmomiroski/todo_app.git"
+	},
+	"bugs": {
```
Comment (AlexMelov) [inline package.json:16]: { on new line

File: package.json [hunk 45]
```diff
@@ -0,0 +1,27 @@
...
+		"url": "https://github.com/vladimirmomiroski/todo_app/issues"
+	},
+	"workspaces": [
```
Comment (AlexMelov) [inline package.json:19]: [ on new line

File: .gitignore (hunk 1 above)
Comment (vladimirmomiroski) [reply .gitignore:6]: done

File: backend/.eslintrc (hunk 4 above)
Comment (vladimirmomiroski) [reply backend/.eslintrc]: done

File: backend/package.json (hunk 8 above)
Comment (vladimirmomiroski) [reply backend/package.json:18]: done

File: backend/src/Controllers/deleteTodoHandler.js (hunk 9 above)
Comment (vladimirmomiroski) [reply backend/src/Controllers/deleteTodoHandler.js]: done

File: backend/src/Controllers/updateTodoHandler.js [hunk 46]
```diff
@@ -0,0 +1,20 @@
...
+      { $set: { isCompleted: true } }
+		);
+
+    response.json(updatedTodo);
+	}
+	catch (error) 
+	{
+    response.json({ message: error });
+	}
+};
```
[the remark this answers is not in the document]
Comment (vladimirmomiroski) [reply backend/src/Controllers/updateTodoHandler.js]: done

File: backend/src/Routes/routes.js (hunk 10 above)
Comment (vladimirmomiroski) [reply backend/src/Routes/routes.js:1]: done

File: backend/src/index.js [hunk 47]
```diff
@@ -0,0 +1,30 @@
...
+// Middlewares
+app.use(cors());
+
+// routes
+
+app.use('/', require('./Routes/routes'));
+app.use('/todos', require('./Routes/routes'));
+app.use('/todos/:id', require('./Routes/routes'));
+
+// db connection
```
[the remark this answers is not in the document]
Comment (vladimirmomiroski) [reply backend/src/index.js:18]: done

File: backend/src/index.js (hunk 13 above)
Comment (vladimirmomiroski) [reply backend/src/index.js:29]: done

File: backend/src/models/Todos.js (hunk 14 above)
Comment (vladimirmomiroski) [reply backend/src/models/Todos.js:3]: done

File: backend/src/models/Todos.js (hunk 15 above)
Comment (vladimirmomiroski) [reply backend/src/models/Todos.js:4]: done

File: backend/src/models/Todos.js (hunk 16 above)
Comment (vladimirmomiroski) [reply backend/src/models/Todos.js:8]: done

File: frontend/.eslintrc (hunk 18 above)
Comment (vladimirmomiroski) [reply frontend/.eslintrc:7]: done

File: frontend/.eslintrc (hunk 20 above)
Comment (vladimirmomiroski) [reply frontend/.eslintrc:15]: done

File: frontend/package.json (hunk 21 above)
Comment (vladimirmomiroski) [reply frontend/package.json:5]: done

File: frontend/package.json (hunk 23 above)
Comment (vladimirmomiroski) [reply frontend/package.json:21]: done

File: frontend/package.json (hunk 24 above)
Comment (vladimirmomiroski) [reply frontend/package.json:27]: done

File: frontend/package.json (hunk 25 above)
Comment (vladimirmomiroski) [reply frontend/package.json:32]: done

File: frontend/package.json (hunk 26 above)
Comment (vladimirmomiroski) [reply frontend/package.json]: done

File: frontend/package.json (hunk 27 above)
Comment (vladimirmomiroski) [reply frontend/package.json]: done

File: frontend/src/App.js (hunk 29 above)
Comment (vladimirmomiroski) [reply frontend/src/App.js:5]: done

File: frontend/src/Todo/Body/BodyItemActions.jsx (hunk 30 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/Body/BodyItemActions.jsx:14]: done

File: frontend/src/Todo/Context.js (hunk 32 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/Context.js:36]: done

File: frontend/src/Todo/Context.js (hunk 33 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/Context.js:40]: done

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js (hunk 34 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:19]: done

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js (hunk 35 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:21]: done

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js (hunk 36 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:43]: done

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js (hunk 37 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:45]: done

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js (hunk 38 above)
Comment (vladimirmomiroski) [reply frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:53]: done

File: .gitignore [hunk 48]
```diff
@@ -1,18 +1,16 @@
-# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
+.idea
```
Comment (henryruhs) [inline .gitignore:1]: Copy from Alek's project:

```
.idea
.vscode
.env
.eslintcache
.stylelintcache
.angular
.vercel
build
dist
node_modules
```

File: README.md [hunk 49]
```diff
@@ -1 +1,65 @@
...
+
+```
+REACT_APP_TODOS_URL=http://localhost:5000/todos
+```
+
+Create a .env file inside backend directory:
```
Comment (henryruhs) [inline README.md:30]: you can also use ` for to highlight the file

File: README.md [hunk 50]
```diff
@@ -1 +1,65 @@
...
+```
+http://localhost:3000
+```
+
+Browse your backend api on:
+
+```
+http://localhost:5000/todos
+```
+
```
Comment (henryruhs) [inline README.md:59]: Remove empty lines at the end

File: backend/.eslintrc [hunk 51]
```diff
@@ -0,0 +1,22 @@
...
+    "es6": true,
+    "node": true
+  },
+  "extends": ["@isnotdefined/eslint-config"],
+  "parserOptions": 
+  {
+    "sourceType": "module",
+    "ecmaVersion": 2018
+  },
+
```
Comment (henryruhs) [inline backend/.eslintrc:13]: Remove empty line

File: backend/package.json [hunk 52]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "backend",
+  "version": "1.0.0",
+  "main": "App.js",
+  "dependencies": 
+  {
+    "config": "3.3.1",
+    "cors": "^2.8.5",
+    "dotenv": "^16.0.0",
```
Comment (henryruhs) [inline backend/package.json:9]: I think dotenv is a devDependency?

File: backend/src/Controllers/addTodoHandler.js [hunk 53]
```diff
@@ -0,0 +1,22 @@
+const Todo = require('../Models/Todos');
+
+exports.addTodoHandler = (request, response) => 
+{
+	const todo = request.body;
+
```
Comment (henryruhs) [inline backend/src/Controllers/addTodoHandler.js:6]: Remove empty line between const

File: backend/src/Controllers/updateTodoHandler.js [hunk 54]
```diff
@@ -0,0 +1,19 @@
...
+
+	Todo.updateOne(
+      { _id: id },
+      { $set: { isCompleted: true } }
+	  ) 
+	  .then(data => 
+	  {
+		response.json(data);
+	  })
+	  .catch(error =>
```
Comment (henryruhs) [inline backend/src/Controllers/updateTodoHandler.js:15]: You can use a shortcut like .catch(error => throw new Error(error));

File: backend/src/Routes/routes.js [hunk 55]
```diff
@@ -0,0 +1,17 @@
...
+
```
Comment (henryruhs) [inline backend/src/Routes/routes.js:2]: Rename file to server.js

File: frontend/.eslintrc [hunk 56]
```diff
@@ -0,0 +1,27 @@
+{
+	"env": 
+	{
+		"es6": true,
+		"node": true
+	},
+	"extends": ["@isnotdefined/eslint-config", "react-app", "react-app/jest"],
```
Comment (henryruhs) [inline frontend/.eslintrc:7]: Please use just this, thanks:

```
"extends":
[
    "@isnotdefined/eslint-config",
    "react-app",
    "react-app/jest"
],
```

File: frontend/.eslintrc [hunk 57]
```diff
@@ -0,0 +1,27 @@
...
+	"parserOptions": 
+	{
+		"sourceType": "module",
+		"ecmaFeatures": {
+			"jsx": true
+		},
+		"ecmaVersion": 2018
+	},
+
```
Comment (henryruhs) [inline frontend/.eslintrc:16]: Remove empty line

File: frontend/package.json [hunk 58]
```diff
@@ -2,34 +2,48 @@
...
-    ]
+    "eject": "react-scripts eject",
+    "lint": "eslint src && stylelint src",
+    "fix": "eslint src --fix && stylelint src --fix"
   },
-  "browserslist": {
-    "production": [
+  "browserslist": 
+  {
+    "production":
```
Comment (henryruhs) [inline frontend/package.json:39]: Move to .browserlistrc file ... and just last 2 versions

File: frontend/src/Todo/Body/Body.jsx [hunk 59]
```diff
@@ -0,0 +1,21 @@
+import { useContext } from 'react';
+import classes from './Body.module.css';
+import { Context } from '../Context';
+import BodyItem from './BodyItem';
+
+export default function Body() 
+{
+	const { container, ul } = classes;
+
```
Comment (henryruhs) [inline frontend/src/Todo/Body/Body.jsx:9]: Remove empty line

File: frontend/src/Todo/Body/BodyItem.jsx [hunk 60]
```diff
@@ -0,0 +1,17 @@
+import classes from './BodyItem.module.css';
+import BodyItemCheckbox from './BodyItemCheckbox';
+import BodyItemActions from './BodyItemActions';
+
+export default function TodoBodyItem({ item }) 
+{
+	const { li } = classes;
+
```
Comment (henryruhs) [inline frontend/src/Todo/Body/BodyItem.jsx:8]: Remove empty line

File: frontend/src/Todo/Body/BodyItemActions.jsx [hunk 61]
```diff
@@ -0,0 +1,18 @@
+import { useContext } from 'react';
+import classes from './BodyItemActions.module.css';
+import { Context } from '../Context';
+
+export default function TodoBodyItemActions({ id }) 
+{
+	const { deleteBtn } = classes;
+
```
Comment (henryruhs) [inline frontend/src/Todo/Body/BodyItemActions.jsx:8]: Delete empty line

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 62]
```diff
@@ -0,0 +1,66 @@
+const urlToFetch = process.env.REACT_APP_TODOS_URL;
+
+export const fetchData = setData => 
+{
+    fetch(`${urlToFetch}`)
```
Comment (henryruhs) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:5]: Use `fetch(url)` please

File: frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js [hunk 63]
```diff
@@ -0,0 +1,66 @@
...
+    .then(res => res.json())
+    .then(dataFromServer => 
+    {
+      	setData(dataFromServer);
+    })
+    .catch(error =>
```
Comment (henryruhs) [inline frontend/src/Todo/FetchMethodsHandler/FetchMethodsHandler.js:11]: catch(error => throw new Error(error));

File: frontend/src/Todo/Header/Header.jsx [hunk 64]
```diff
@@ -0,0 +1,42 @@
+import { useContext, useState } from 'react';
+import classes from './Header.module.css';
+import { Context } from '../Context';
+
+export default function TodoHeader() 
+{
+	const { container, input, btn } = classes;
+
```
Comment (henryruhs) [inline frontend/src/Todo/Header/Header.jsx:8]: Remove empty lines between const

File: frontend/src/Todo/Header/Header.jsx [hunk 65]
```diff
@@ -0,0 +1,42 @@
...
+	const { addTodo } = useContext(Context);
+
+	const [ todoName, setTodoName ] = useState('');
+
+	const onSubmitHandler = e => 
+	{
+   		e.preventDefault();
+
+    	if (todoName)
```
Comment (henryruhs) [inline frontend/src/Todo/Header/Header.jsx:17]: Is this file linted? You are using spaces...

File: frontend/src/index.css [hunk 66]
```diff
@@ -1,13 +1,7 @@
...
-    sans-serif;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-}
+@import '../node_modules/harmonize.css/harmonize.css';
 
-code {
-  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
-    monospace;
+code
```
Comment (henryruhs) [inline frontend/src/index.css:3]: Do we even use code tag?

**Review by henryruhs — APPROVED** [review]

File: frontend/src/Todo/Body/BodyItem.jsx [hunk 67]
```diff
@@ -0,0 +1,17 @@
+import classes from './BodyItem.module.css';
```
Comment (henryruhs) [inline frontend/src/Todo/Body/BodyItem.jsx:1]: Rename classes to styles

File: frontend/src/Todo/Body/BodyItemActions.jsx [hunk 68]
```diff
@@ -0,0 +1,18 @@
+import { useContext } from 'react';
+import classes from './BodyItemActions.module.css';
```
Comment (henryruhs) [inline frontend/src/Todo/Body/BodyItemActions.jsx:2]: Rename classes to styles

File: frontend/src/Todo/Body/BodyItemCheckbox.module.css [hunk 69]
```diff
@@ -0,0 +1,21 @@
+.container
+{
+	align-items: center;
+	display: flex;
+}
+
+.nameText
```
Comment (henryruhs) [inline frontend/src/Todo/Body/BodyItemCheckbox.module.css:7]: CSS is hyphen based - not camelCase... .text-name

File: frontend/src/Todo/Header/Header.module.css [hunk 70]
```diff
@@ -0,0 +1,34 @@
...
+	background-color: #fff;
+	border: 0.2rem  solid #adefd1;
+	border-radius: 0.2rem;
+	color: #00203f;
+	cursor: pointer;
+	font-weight: 700;
+	padding: 0.5rem 1.5rem;
+}
+
+.btn:hover
```
Comment (henryruhs) [inline frontend/src/Todo/Header/Header.module.css:30]: button:hover

File: frontend/src/Todo/Header/Header.module.css [hunk 71]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+.input
+{
+	border: 0.2rem solid #adefd1;
+	border-radius: 0.2rem;
+	padding: 0.6rem 1rem 0.6rem 0.4rem;
+}
+
+.btn
```
Comment (henryruhs) [inline frontend/src/Todo/Header/Header.module.css:19]: .button

**Comment by vladimirmomiroski** [comment]
All changes are merged into master branch

**Pull request merged**